stdlib/public/core/Span/Span.swift

```diff
@@ -12,236 +12,270 @@
   }
 
-  /// Unsafely create a `Span` over initialized memory.
+  /// Unsafely creates a `Span` over initialized memory.
   ///
-  /// The memory in `buffer` must be owned by the instance `owner`,
-  /// meaning that as long as `owner` is alive the memory will remain valid.
+  /// The memory in `buffer` must remain valid throughout the lifetime of
+  /// the newly-created `Span`.
   ///
   /// - Parameters:
   ///   - buffer: an `UnsafeMutableBufferPointer` to initialized elements.
   ///   - owner: a binding whose lifetime must exceed that of
   ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
   @_alwaysEmitIntoClient
+  @lifetime(immortal)
   public init(
-    _unsafeElements buffer: UnsafeMutableBufferPointer<Element>
-  ) -> dependsOn(immortal) Self {
-    self.init(_unsafeElements: UnsafeBufferPointer(buffer))
+    _unsafeElements buffer: Slice<UnsafeMutableBufferPointer<Element>>
+  ) {
+    self.init(_unsafeElements: UnsafeBufferPointer(rebasing: buffer))
   }
+}
 
-  /// Unsafely create a `Span` over initialized memory.
-  ///
-  /// The memory representing `count` instances starting at
-  /// `pointer` must be owned by the instance `owner`,
-  /// meaning that as long as `owner` is alive the memory will remain valid.
-  ///
-  /// - Parameters:
-  ///   - pointer: a pointer to the first initialized element.
-  ///   - count: the number of initialized elements in the span.
-  ///   - owner: a binding whose lifetime must exceed that of
-  ///            the newly created `Span`.
-  @_alwaysEmitIntoClient
-  public init(
-    _unsafeStart start: UnsafePointer<Element>,
-    count: Int
-  ) -> dependsOn(immortal) Self {
-    precondition(count >= 0, "Count must not be negative")
-    self.init(_unchecked: start, count: count)
-  }
+@_disallowFeatureSuppression(NonescapableTypes)
+extension Span where Element: BitwiseCopyable {
 
-  /// Unsafely create a `Span` over initialized memory.
+  /// Unsafely creates a `Span` over initialized memory.
   ///
-  /// The memory in `unsafeBytes` must be owned by the instance `owner`
-  /// meaning that as long as `owner` is alive the memory will remain valid.
+  /// The memory in `buffer` must remain valid throughout the lifetime of
+  /// the newly-created `Span`.
   ///
-  /// `unsafeBytes` must be correctly aligned for accessing
+  /// `buffer` must be correctly aligned for accessing
   /// an element of type `Element`, and must contain a number of bytes
   /// that is an exact multiple of `Element`'s stride.
   ///
   /// - Parameters:
-  ///   - unsafeBytes: a buffer to initialized elements.
+  ///   - buffer: a buffer to initialized elements.
   ///   - type: the type to use when interpreting the bytes in memory.
   ///   - owner: a binding whose lifetime must exceed that of
   ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
   @_alwaysEmitIntoClient
+  @lifetime(immortal)
```

Comment:
Should be `@lifetime(buffer)`. The resulting Span depends on the validity of the buffer.